book/src/development/define_lints.md

```diff
@@ -0,0 +1,177 @@
+# Define New Lints
+
+The journey of any new Clippy lint begins with one step.
+And this first step is to define the lint and register it to the codebase.
+We can use the Clippy dev tools to handle this step since setting up the 
+lint involves some boilerplate code.
+
+## Name the Lint
+
+A wise software engineer once said:
+> There are only two hard things in Computer Science: cache invalidation and naming things.
```

Comment:
You forgot "...and off by one errors." 😁

Reply:
I'm using his son's blog haha: https://www.karlton.org/2017/12/naming-things-hard/